src/core/jit/exec.rs

```diff
@@ -97,6 +97,51 @@ where
         .await;
     }
 
+    #[allow(clippy::too_many_arguments)]
+    async fn execute_inner<'b>(
+        &'b self,
+        field: &'b Field<Nested<Input>, Input>,
+        ctx: &'b Context<'b, Input, Output>,
+        data_path: DataPath,
+        value: Option<&'b Output>,
+    ) -> Result<(), Error> {
+        if field.type_of.is_list() {
+            // Check if the value is an array
+            if let Some(array) = value.and_then(|v| v.as_array()) {
+                join_all(field.nested_iter().map(|field| {
+                    join_all(array.iter().enumerate().map(|(index, value)| {
+                        let new_value = value.get_key(&field.name).unwrap_or(value);
+                        let ctx = ctx.with_value_and_field(new_value, field);
+                        let data_path = data_path.clone().with_index(index);
+                        async move { self.execute(field, &ctx, data_path).await }
+                    }))
+                }))
+                .await;
+            }
+            // TODO:  We should throw an error stating that we expected
+            // a list type here but because the `Error` is a
+            // type-parameter, its not possible
+        }
+        // TODO: Validate if the value is an Object
+        // Has to be an Object, we don't do anything while executing if its a Scalar
+        else {
+            join_all(field.nested_iter().map(|child| {
+                let value = if value.is_some() { value } else { ctx.value() };
```

Comment:
this line is redundant because of the way you're calling this function from `exec`